### DISCUSSION TITLE
Will running multiple chatbots from the same ip lead to account being blocked?

### DISCUSSION THREAD

**Discussion by linhandev** [acheong08/ChatGPT#884] [Q&A] [answered]
Hope guys who have this experience can share some insights.

**Comment by acheong08** [comment, marked as answer]
There is a rate limit of 50 requests / minute per IP

**Reply by acheong08** [reply]
If you have a dedicated need, you can email me for a non-rate limited server

**Reply by linhandev** [reply]
Many thanks! Thant's very kind! I'm aware there is this 50 req/min limit on the duti reverse proxy server and haven't run into this limit yet. 

What I meant to ask is, say someone is operating a pool of bots on a single server, there will be
- login to multiple accounts, possibly in short succession during pool init
- chatbot.ask with different account credentials

from the same ip address. I'm wondering will these or other things I haven't thought of lead to the accounts being banned.

It seems to me during chatbot.ask, all traffic is peroxyed through the duti proxy. This makes me think on the openai part, they will only see the requests coming from the proxy server, not the bot pool server. So the second case may not matter. Is this a correct conclusion?

I've went through the code and I think the login process isn't behind any proxy by default. This way I think it's possible for openai to see multiple logins to different accounts in short succession from the same ip and mark them. Have you heard of anyone being banned for this?

Many thanks again for the great project.

**Reply by acheong08** [reply]
> login to multiple accounts, possibly in short succession during pool init

You may get rate limited by auth0 since that is done locally 

> chatbot.ask with different account credentials

Yes, this does not get you rate limited by OpenAI

**Reply by linhandev** [reply]
Copy that. Thanks.

**Reply by linhandev** [reply]
I ran into login rate limit today. error code in auth package is 429

@acheong08

**Reply by acheong08** [reply]
Will update. Which part was it at?

**Reply by linhandev** [reply]
![image](https://user-images.githubusercontent.com/29757093/221446243-05523976-a8e6-4f58-994d-58158e311de7.png)

raised here. I'm actually not sure if response.status_code is 429, but the response.text is a webpage like the following. Terminal log got flushed and I currently can login

![image](https://user-images.githubusercontent.com/29757093/221446366-e87088fb-daee-4868-9821-e8178a64f859.png)
